x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1174,29 +1175,26 @@ Collection<Object> createComponents(
         components.add(ipFilter.get());
 
         DestructiveOperations destructiveOperations = new DestructiveOperations(settings, clusterService.getClusterSettings());
-        crossClusterAccessAuthcService.set(new CrossClusterAccessAuthenticationService(clusterService, apiKeyService, authcService.get()));
-        components.add(crossClusterAccessAuthcService.get());
 
-        RemoteClusterTransportInterceptor remoteClusterTransportInterceptor = new CrossClusterAccessTransportInterceptor(
-            settings,
-            threadPool,
+        RemoteClusterSecurityExtension.Components rcsComponents = new RemoteClusterSecurityComponents(
             authcService.get(),
             authzService,
             securityContext.get(),
-            crossClusterAccessAuthcService.get(),
-            getLicenseState()
-        );
-
-        var crossClusterApiKeySignerReloader = new CrossClusterApiKeySigningConfigReloader(
-            environment,
+            apiKeyService,
             resourceWatcherService,
-            clusterService.getClusterSettings()
+            projectResolver,
+            getLicenseState(),
+            clusterService,
+            environment,
+            threadPool,
+            settings,
+            client
         );
-        components.add(crossClusterApiKeySignerReloader);
-
-        var crossClusterApiKeySigner = new CrossClusterApiKeySigner(environment);
-        crossClusterApiKeySignerReloader.setApiKeySigner(crossClusterApiKeySigner);
-        components.add(crossClusterApiKeySigner);
+        remoteClusterSecurityExtension.set(this.getRemoteClusterSecurityExtension(rcsComponents));
+        remoteClusterAuthenticationService.set(remoteClusterSecurityExtension.get().getAuthenticationService());
+        components.add(new PluginComponentBinding<>(RemoteClusterAuthenticationService.class, remoteClusterAuthenticationService.get()));
```

Comment:
Why do we need to use a binding here?

Reply:
Looking closely, it's not necessary to expose authentication service at all as a component.  
I mostly kept it since it was exposed [previously](https://github.com/elastic/elasticsearch/blob/e6d78b01c081f5a600a139159a72be1714c19054/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java#L1178).

The binding was intended to support injecting service by its interface type (`RemoteClusterAuthenticationService`) instead of a concrete implementation (`CrossClusterAccessAuthenticationService`).